src/ott/geometry/geometry.py

```diff
@@ -129,30 +129,58 @@ def mean_cost_matrix(self) -> float:
     tmp = self._masked_geom().apply_cost(self._n_normed_ones).squeeze()
     return jnp.sum(tmp * self._m_normed_ones)
 
+  @property
+  def std_cost_matrix(self) -> float:
+    r"""Standard deviation of all values stored in :attr:`cost_matrix`.
+
+    Uses the :meth:`~ott.geometry.Geometry.apply_square_cost` to remain
+    applicable to low-rank matrices, through the formula:
+
+    .. math::
+        \sigma^2=\frac{1}{nm}\left(\sum_{ij} C_{ij}^2 -
+        (\sum_{ij}C_ij)^2\right).
+
+    to output :math:`\sigma`.
+    """
+    tmp = self._masked_geom().apply_square_cost(self._n_normed_ones).squeeze()
+    return jnp.sqrt(
+        jax.nn.
+        relu(jnp.sum(tmp * self._m_normed_ones) - (self.mean_cost_matrix) ** 2)
+    )
+
   @property
   def kernel_matrix(self) -> jnp.ndarray:
     """Kernel matrix.
 
     Either provided by user or recomputed from :attr:`cost_matrix`.
     """
     if self._kernel_matrix is None:
-      return jnp.exp(-(self._cost_matrix * self.inv_scale_cost / self.epsilon))
+      return jnp.exp(-self._cost_matrix * self.inv_scale_cost / self.epsilon)
     return self._kernel_matrix ** self.inv_scale_cost
 
   @property
   def _epsilon(self) -> epsilon_scheduler.Epsilon:
     (target, scale_eps, _, _), _ = self._epsilon_init.tree_flatten()
     rel = self._relative_epsilon
 
-    use_mean_scale = rel is True or (rel is None and target is None)
-    if scale_eps is None and use_mean_scale:
-      scale_eps = jax.lax.stop_gradient(self.mean_cost_matrix)
+    # If nothing passed, default to STD
+    if (rel is None and target is None and scale_eps is None):
+      scale_eps = jax.lax.stop_gradient(self.std_cost_matrix)
+    # If instructions passed change, otherwise (notably if False) skip.
+    elif rel is not None:
+      if rel == "mean" or rel is True:  # Legacy option.
+        scale_eps = jax.lax.stop_gradient(self.mean_cost_matrix)
+      elif rel == "std":
+        scale_eps = jax.lax.stop_gradient(self.std_cost_matrix)
+        # Avoid 0 std, since this would set epsilon to 0.0 and result in
+        # a division by 0.
+        scale_eps = jnp.where(scale_eps <= 0.0, 1.0, scale_eps)
```

Comment:
Would use `jax.nn.relu(scale_eps)`.

Reply:
that's not the same :)

Reply:
Ah sorry, completely missed it, thank you!